src/Microsoft.Health.Dicom.Core.UnitTests/Serialization/Newtonsoft/ConfigurationJsonConverterTests.cs

```diff
@@ -18,17 +18,106 @@ public class ConfigurationJsonConverterTests
 
     public ConfigurationJsonConverterTests()
     {
-        _serializerSettings = new JsonSerializerSettings { Formatting = Formatting.Indented };
+        _serializerSettings = new JsonSerializerSettings
+        {
+            Formatting = Formatting.Indented,
+            DateFormatHandling = DateFormatHandling.IsoDateFormat,
+            DateParseHandling = DateParseHandling.None,
+            DateTimeZoneHandling = DateTimeZoneHandling.Utc,
+        };
         _serializerSettings.Converters.Add(new ConfigurationJsonConverter(new CamelCaseNamingStrategy()));
     }
 
+    [Fact]
+    public void GivenJson_WhenReadingWithNoDateParsing_ThenDeserialize()
+    {
+        _serializerSettings.DateParseHandling = DateParseHandling.None;
+
+        const string json = @"{
+  ""date"": ""2022-04-14T23:39:26.7818757Z"",
+  ""dateOffset"": ""2022-05-03T15:35:57.2628853-03:00"",
+}";
+
+        IConfiguration actual = JsonConvert.DeserializeObject<IConfiguration>(json, _serializerSettings);
+
+        Assert.Equal("2022-04-14T23:39:26.7818757Z", actual["date"]);
+        Assert.Equal("2022-05-03T15:35:57.2628853-03:00", actual["dateOffset"]);
+    }
+
+    [Fact]
+    public void GivenJson_WhenReadingWithDateTimeParsing_ThenDeserialize()
+    {
+        _serializerSettings.DateParseHandling = DateParseHandling.DateTime;
+
+        const string json = @"{
+  ""date"": ""2022-04-14T23:39:26.7818757Z"",
+  ""dateOffset"": ""2022-05-03T15:35:57.2628853-03:00"",
```

Comment:
It could be dangerous to use the `DateTime` type for parsing, but luckily for our Durable Task code we specify "None" so that Json.NET does not try to parse the object prematurely